server/datastore/mysql/software_installers_test.go

```diff
@@ -28,6 +32,145 @@ func TestSoftwareInstallers(t *testing.T) {
 	}
 }
 
+func testListSoftwareInstallerDetails(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+
+	host1 := test.NewHost(t, ds, "host1", "1", "host1key", "host1uuid", time.Now())
+	host2 := test.NewHost(t, ds, "host2", "2", "host2key", "host2uuid", time.Now())
+
+	script1, err := insertScriptContents(ctx, "hello", ds.writer(ctx))
+	require.NoError(t, err)
+	script1Id, err := script1.LastInsertId()
+	require.NoError(t, err)
+
+	script2, err := insertScriptContents(ctx, "world", ds.writer(ctx))
+	require.NoError(t, err)
+	script2Id, err := script2.LastInsertId()
+	require.NoError(t, err)
+
+	installer1, err := insertSoftwareInstaller(ctx, ds.writer(ctx), "file1", "1.0", "SELECT 1", "storage1", script1Id, script2Id)
+	require.NoError(t, err)
+	installer1Id, err := installer1.LastInsertId()
+	require.NoError(t, err)
+
+	installer2, err := insertSoftwareInstaller(ctx, ds.writer(ctx), "file2", "2.0", "SELECT 2", "storage2", script2Id, script1Id)
+	require.NoError(t, err)
+	installer2Id, err := installer2.LastInsertId()
+	require.NoError(t, err)
+
+	hostInstall1, err := insertHostSoftwareInstalls(ctx, ds.writer(ctx), host1.ID, "exec1", uint(installer1Id))
+	require.NoError(t, err)
+	_ = hostInstall1
+
+	hostInstall2, err := insertHostSoftwareInstalls(ctx, ds.writer(ctx), host1.ID, "exec2", uint(installer2Id))
+	require.NoError(t, err)
+	_ = hostInstall2
+
+	hostInstall3, err := insertHostSoftwareInstalls(ctx, ds.writer(ctx), host2.ID, "exec3", uint(installer1Id))
+	require.NoError(t, err)
+	_ = hostInstall3
+
+	hostInstall4, err := insertHostSoftwareInstalls(ctx, ds.writer(ctx), host2.ID, "exec4", uint(installer2Id))
+	require.NoError(t, err)
+	hostInstall4Id, err := hostInstall4.LastInsertId()
+	require.NoError(t, err)
+
+	_ = ds.writer(ctx).MustExec("UPDATE host_software_installs SET install_script_exit_code = 0 WHERE id = ?", hostInstall4Id)
```

Comment:
I'd suggest adding cases where it failed pre-condition, install and post-condition (in addition to the true pending state), just to cover all bases.

Reply:
done